cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -231,30 +231,105 @@ void AddColumnIndices(const SchemaField& schema_field,
   }
 }
 
-// Compute the column projection out of an optional arrow::Schema
-std::vector<int> InferColumnProjection(const parquet::arrow::FileReader& reader,
-                                       const ScanOptions& options) {
+Status ResolveOneFieldRef(
+    const SchemaManifest& manifest, const FieldRef& field_ref,
+    const std::unordered_map<std::string, const SchemaField*>& field_lookup,
+    const std::unordered_set<std::string>& duplicate_fields,
+    std::vector<int>* columns_selection) {
+  if (const auto* name = field_ref.name()) {
```

Comment:
```suggestion
  if (const std::string* name = field_ref.name()) {
```
Same optional nit as above.